python_modules/dagster/dagster/_core/definitions/sensor_definition.py

```diff
@@ -1030,9 +1031,11 @@ def _get_repo_job_by_name(context: SensorEvaluationContext, job_name: str) -> Jo
                 *_run_requests_with_base_asset_jobs(run_requests, context, asset_selection)
             ]
 
-        dynamic_partitions_store = (
-            CachingDynamicPartitionsLoader(context.instance) if context.instance_ref else None
-        )
+        @functools.cache
+        def dynamic_partitions_store():
+            return (
+                CachingDynamicPartitionsLoader(context.instance) if context.instance_ref else None
+            )
```

Comment:
`@functools.cache` is 3.9+ and we haven't quite dropped 3.8 support yet 

I think a memoized getter (`@cached_property` or manual) on `SensorEvaluationContext` is a bit more straight forward than this cached inline function

Reply:
ugh i keep forgetting about that. ok thanks